service/src/main/java/org/cbioportal/genome_nexus/service/internal/HotspotFilter.java

```diff
@@ -95,15 +95,18 @@ else if (hotspotType.contains("3d"))  {
             // Protein location
             int hotspotStart = proteinPositionResolver.extractProteinPos(hotspot.getResidue()).getStart();
             int hotspotStop = proteinPositionResolver.extractProteinPos(hotspot.getResidue()).getEnd();
-            validPosition &= (start <= hotspotStart && end >= hotspotStop);
+
+            // as long as there is overlap, it should be valid position
+            validPosition &= (start <= hotspotStop && end >= hotspotStart);
```

Comment:
I know that we previously approved this but actually `&=` is confusing here, we should just remove the line where we initialize this boolean (` boolean validPosition = true;`) and change this line to:
```suggestion
            boolean validPosition = start <= hotspotStop && end >= hotspotStart;
```